src/yggdrasil/dialer.go

```diff
@@ -21,7 +21,7 @@ type Dialer struct {
 // "curve25519" or "nodeid" and the second parameter should contain a
 // hexadecimal representation of the target. It uses DialContext internally.
 func (d *Dialer) Dial(network, address string) (net.Conn, error) {
-	return d.DialContext(nil, network, address)
+	return d.DialContext(context.TODO(), network, address)
```

Comment:
As mentioned in the yggdrasilsim code, we should remove the `ctx == nil` check elsewhere in the dial code, and use `context.Background()` here instead, I think.